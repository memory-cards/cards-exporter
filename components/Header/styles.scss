body {
  margin: 0;

  font-family: "Arimo", sans-serif;
}

// .nav {
//   height: 40px;
//   display: flex;
//   justify-content: space-between;
//   background-color: lightgrey;
//   margin-bottom: 20px;

//   .navbar {
//     display: flex;

//     & a {
//       display: flex;
//       align-items: center;
//       text-decoration: none;
//       padding: 0 10px;
//       font-weight: bold;
//       color: black;

//       &:hover {
//         background-color: #c2bfbf;
//       }

//       &.active {
//         background-color: #bea3a3;
//       }
//     }
//   }
// }
