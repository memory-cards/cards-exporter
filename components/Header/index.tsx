// tslint:disable no-submodule-imports
import Link from "next/link";
import { withRouter, WithRouterProps } from "next/router";
import * as React from "react";

import "bootstrap/dist/css/bootstrap.min.css";

import "./styles.scss";

const pages = [
  { href: "/filterCards", label: "Filter cards page" },
  { href: "/createCard", label: "Create cards page" },
  { href: "/updateCards", label: "Update cards page" }
];

const infoLinks = [
  { href: "/api/date", label: "Date info" },
  { href: "/api/cards/list", label: "Cards info" },
  { href: "/api/cards/deck", label: "Deck info" },
  { href: "/api/cards/update", label: "Last update info" },
  { href: "/api/cards/tags", label: "Tags info" }
];

const Header = (props: WithRouterProps) => {
  return (
    <div className="nav">
      <div className="navbar">
        {infoLinks.map(({ href, label }) => (
          <Link key={href} href={href}>
            <a>{label}</a>
          </Link>
        ))}
      </div>

      <div className="navbar">
        {pages.map(({ href, label }) => (
          <Link key={href} href={href}>
            <a
              className={
                props && props.router && href === props.router.pathname
                  ? "active"
                  : ""
              }
            >
              {label}
            </a>
          </Link>
        ))}
      </div>
    </div>
  );
};

export { Header as TestHeader };

export default withRouter(Header);
