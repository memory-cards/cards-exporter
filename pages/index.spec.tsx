import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";
import * as TestRenderer from "react-test-renderer";

import Header from "../components/Header";
import Index from "./index";

describe("With Enzyme", () => {
  let page: ShallowWrapper;

  beforeEach(() => {
    page = shallow(<Index />);
  });

  describe("Main page", () => {
    it('has title "Cards Exporter"', () => {
      expect(page.find("h1").text()).toEqual("Cards Exporter");
    });

    it("renders Header", () => {
      expect(page.find(Header).exists()).toBe(true);
    });
  });
});

describe("With Snapshot Testing", () => {
  it('IndexPage shows "Cards Exporter"', () => {
    const component = TestRenderer.create(<Index />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
