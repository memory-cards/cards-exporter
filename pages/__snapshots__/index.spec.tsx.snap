// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing IndexPage shows "Cards Exporter" 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/api/date"
        onClick={[Function]}
      >
        Date
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/api/cards/list"
        onClick={[Function]}
      >
        Cards
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/api/cards/deck"
        onClick={[Function]}
      >
        Deck
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/api/cards/update"
        onClick={[Function]}
      >
        Last update
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/api/cards/tags"
        onClick={[Function]}
      >
        Tags
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/filterCards"
        onClick={[Function]}
      >
        Filter cards
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/createCard"
        onClick={[Function]}
      >
        Create cards
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/updateCards"
        onClick={[Function]}
      >
        Update cards
      </a>
    </li>
  </ul>
  <h1>
    Cards Exporter
  </h1>
</div>
`;
